src/DateTimePicker.jsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import cx from 'classnames';
+import Flatpickr from './FlatpickrComponent';
+import TimeInput from './TimeInput';
+import DateTimeLocalInput from './DateTimeLocalInput';
+
+/**
+ * @module DateTimePicker
+ * a component that renders a calendar ui and time input
+ * defaults to datetime-local on supported mobile browsers
+ */
+class DateTimePicker extends React.Component {
+
+	constructor(props) {
+		console.log(props.value);
+		super(props);
+		this.state = {
+			datetime: props.value ? new Date(props.value).toISOString() : '',
```

Comment:
you're initializing `state.datetime` to a string, but `setTime` will save it as a `Date` object - it should always be one or the other, and the `Date` object is going to be easier to work with.